panda/src/gobj/texture.cxx

```diff
@@ -10077,8 +10084,17 @@ do_write_datagram_rawdata(CData *cdata, BamWriter *manager, Datagram &me) {
     me.add_uint8(cdata->_ram_images.size());
     for (size_t n = 0; n < cdata->_ram_images.size(); ++n) {
       me.add_uint32(cdata->_ram_images[n]._page_size);
-      me.add_uint32(cdata->_ram_images[n]._image.size());
-      me.append_data(cdata->_ram_images[n]._image, cdata->_ram_images[n]._image.size());
+
+      // Only write the size of the image data written to bam.
+      if(manager->get_tex_compress_level() == BamEnums::BTC_on) {
+        string raw_image = cdata->_ram_images[n]._image.get_data();
+        string compressed_image = compress_string(raw_image, 6);
+        me.add_uint32(compressed_image.length());
+        me.append_data(compressed_image.data(), compressed_image.length());
+      } else {
+        me.add_uint32(cdata->_ram_images[n]._image.size());
+        me.append_data(cdata->_ram_images[n]._image, cdata->_ram_images[n]._image.size());
+      }
```

Comment:
We should throw a meaningful error if the compression format/level requested is not supported.